src/CarbonAware/src/Interfaces/ICarbonIntensityDataSource.cs

```diff
@@ -16,14 +16,23 @@ public interface ICarbonIntensityDataSource
     double MinSamplingWindow { get; }
 
     /// <summary>
-    /// Gets the carbon intensity for a location and start and end time
+    /// Gets the carbon intensity for locations and start and end time
     /// </summary>
     /// <param name="locations">The locations that should be used for getting emissions data.</param>
     /// <param name="periodStartTime">The start time of the period.</param>
     /// <param name="periodEndTime">The end time of the period.</param>
     /// <returns>A list of emissions data for the given time period.</returns>
     public Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset periodStartTime, DateTimeOffset periodEndTime);
 
+        /// <summary>
```

Comment:
Splitting off the instance where we only pass one location to keep the code cleaner and avoid extra array wrapping in the controller